worker/jobs/bulkUpsertSamplesJob.js

```diff
@@ -54,22 +54,31 @@ module.exports = (job, done) => {
         if (result.isFailed) {
           errors.push(result.explanation);
           return Promise.resolve();
-        } else {
-          successCount++;
-          if (featureToggles.isFeatureEnabled('publishSampleInPromiseChain')) {
-            // Wait for publish to complete before resolving the promise.
-            return publisher.publishSample(result, subHelper.model);
-          }
+        }
 
-          /*
-           * Resolve the promise right away, *before* we actually publish
-           * the sample. Under heavy load, publish a sample can get backed up
-           * because we are looking up the subject info and including that in
-           * the payload of the real-time event.
-           */
-          publisher.publishSample(result, subHelper.model);
-          return Promise.resolve();
+        if (featureToggles.isFeatureEnabled('publishSampleInPromiseChain')) {
+          // Wait for publish to complete before resolving the promise.
+          return publisher
+            .publishSample(result, subHelper.model)
+            .then((sample) => {
+              successCount++;
+              return Promise.resolve(sample);
+            }).catch((err) => {
+              errors.push({ isFailed: true,
+                explanation: err.message, });
+              return Promise.resolve();
+            });
         }
+
```

Comment:
@iamigo Problem happens here, where we don't expect the promise. We can check and count success. Should we change the rule for this job, remove the toggle and always wait for the publish sample promise? Seems that we have a different use case here if we always have to verify success/failure for publishSample.
Make sense?

Reply:
We do not need to consider the success or failure of the publish operation at all when counting success/failure of the sample upsert. The publish is irrelevant.

Do not remove the toggle.